packages/spindle-ui/src/Pagination/hooks/useShowItem.test.ts

```diff
@@ -1,44 +1,601 @@
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4]);
+  });
+
+  it('use show item current 1 total 5', () => {
+    const current = 1;
+    const total = 5;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 5]);
+  });
+
+  it('use show item current 2 total 5', () => {
+    const current = 2;
+    const total = 5;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 5]);
+  });
+
+  it('use show item current 3 total 5', () => {
+    const current = 3;
+    const total = 5;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 5]);
+  });
+
+  it('use show item current 4 total 5', () => {
+    const current = 4;
+    const total = 5;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 5]);
+  });
+
+  it('use show item current 5 total 5', () => {
+    const current = 5;
+    const total = 5;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 5]);
+  });
+
+  it('use show item current 1 total 6', () => {
+    const current = 1;
+    const total = 6;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 6]);
+  });
+
+  it('use show item current 2 total 6', () => {
+    const current = 2;
+    const total = 6;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 2, 3, 4, 6]);
+  });
+
+  it('use show item current 3 total 6', () => {
+    const current = 3;
+    const total = 6;
+
+    const { result } = renderHook(() =>
+      useShowItem({
+        current,
+        total,
+        totalThreshold: TOTAL_THRESHOLD,
+        maxShowItemSize: MAX_SHOW_ITEM_SIZE,
+      }),
+    );
+
+    expect(result.current).toEqual([1, 3, 4, 5, 6]);
```

Comment:
<img width="276" alt="image" src="https://user-images.githubusercontent.com/4308418/204218720-68fe4468-88ae-4e07-ae28-37950d556f35.png">

```suggestion
    expect(result.current).toEqual([1, 2, 3, 4, 6]);
```